### PR TITLE
Update instance size calculation to include flattenable fields

### PR DESCRIPTION
**Pull request by eric98zhang**
Update instance size calculation in `ObjectFieldInfo` to account for
flattenable instance fields.

The layout strategy used is given in issue eclipse/openj9#1387.

Signed-off-by: Eric Zhang <eric99.zhang@gmail.com>

### CODE REVIEW

**Comment by eric98zhang** [comment]
@tajila changes for computing the new instance size, the new field offsets, and the new instance description are all in now

File: runtime/tests/vm/resolvefield_tests.c [hunk 1]
```diff
@@ -149,6 +149,14 @@ createFakeROMClass(U_8 *buffer, UDATA bufferSize, const char *className, testFie
 #undef BUF_ALLOC
 }
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+static J9ClassLoader *
+createFakeClassLoader() {
```
Comment (DanHeidinga) [inline runtime/tests/vm/resolvefield_tests.c]: What's the purpose of this?

Comment (eric98zhang) [reply runtime/tests/vm/resolvefield_tests.c]: I added a `J9ClassLoader *classLoader` parameter to the `fieldOffsetsStartDo()` method in `resolvefield.c`. I intended to write a `createFakeClassLoader()` method to go along with this change for`fieldOffsetsStartDo()` tests in `resolvefield_tests.c`.

**Comment by DanHeidinga** [comment]
@tajila Is this still relevant or should it be closed?

**Comment by tajila** [comment]
Work was completed in another PR

**Pull request closed without merging**